lib/OpenQA/Worker/Cache.pm

```diff
@@ -95,7 +96,9 @@ sub download_asset {
             my ($ua, $tx) = @_;
             my $progress     = 0;
             my $last_updated = time;
-            $tx->req->headers->header('If-None-Match' => qq{$etag}) if $etag;
+            if (-e $asset) {    # Assets might be deleted by a sysadmin
+                $tx->req->headers->header('If-None-Match' => qq{$etag}) if $etag;
```

Comment:
can't you combine these ifs? looks rather strange to have an if to go into one line with another if